server/src/main/java/org/elasticsearch/common/geo/GeoHashUtils.java

```diff
@@ -63,93 +61,19 @@ public static long fromMorton(long morton, int level) {
         return (mFlipped << 4) | level;
     }
 
-    /**
-     * Encode to a geohash string from the geohash based long format
-     */
-    public static String stringEncode(long geoHashLong) {
-        int level = (int) geoHashLong & 15;
-        geoHashLong >>>= 4;
-        char[] chars = new char[level];
-        do {
-            chars[--level] = BASE_32[(int) (geoHashLong & 31L)];
-            geoHashLong >>>= 5;
-        } while (level > 0);
-
-        return new String(chars);
-    }
```

Comment:
Personally, I'd prefer to keep such encoding in a separate method for readability.

Reply:
Not sure I understand, there's still a `stringEncode` method that has the same logic. I only inlined the `lot/lat` conversion to the `ghLong`. The actually used public API remains the same: `stringEncode(lon, lat)`

Reply:
Marking it as ready for merge for now. If we ever need a standalone lat/lon pair to long geo hash encoding we can easily split it up again.